File: Python/Cocaine simulation/cocaine_in_rat_brain.py
def cocaine_in_rat_brain(*arg):
    import numpy as np
    #modellerer coke i hjernen over tiden t i sekunder.... t0 er infusionstid og D er dosis i
    #mg/kg; Data fra PAN et al 1991, Differences in the Pharmacokinetics of Cocaine, J. Neurochem. Modellen er for IV dosering!!!
    t = 0
    t0 = 0
    Dose = 7.5
    if len(arg)==3:
        t = arg[0]
        t0 = arg[1]
        Dose = arg[2]       
    if len(arg)==2:
        t = arg[0]
        t0 = arg[1]
        Dose = 7.5
    if len(arg)<2:
        t = arg[0]
        t0 = 0        
        Dose = 7.5
    if ( len(arg) == 0 or not(type(arg[0])==np.ndarray) ):
        print "Error: Function takes atleast one ndarray argument"
        return 0
    
    
    
    d_v2 = 124*Dose/7.5
    k12 = 0.332/60
    k21 = 0.182/60
    kel = 0.468/60
    
    sumk = k12+k21+kel
    D = np.sqrt(sumk**2 - 4*k21*kel)    
    
    a = 0.5*(sumk + D)
    b = 0.5*(sumk - D)
    
    expA = np.exp( -a*np.array([t[k]-t0 for k in range(0, len(t))]) )
    expB = np.exp( -b*np.array([t[k]-t0 for k in range(0, len(t))]) )
    
    C = d_v2*k12/(a-b) * ( expB - expA ) 
    
    tcoc = 1*(t>t0)
    
    c2 = np.multiply(C,tcoc)
    
    return c2

File: Matlab/Mfiles/cocaine_in_rat_brain.m
function c2 = cocaine_in_rat_brain(t, t0, Dose)
%modellerer coke i hjernen over tiden t i sekunder.... t0 er infusionstid og D er dosis i
%mg/kg; Data fra PAN et al 1991, Differences in the Pharmacokinetics of Cocaine, J. Neurochem. Modellen er for IV dosering!!!

if nargin < 3
    Dose = 7.5;
end
if nargin < 2
    t0 = 0;
end


d_v2 = 124*Dose/7.5;
k12 = 0.332/60;
k21 = 0.182/60;%min^-1 omregnet til s^-1
kel = 0.468/60;
 %\muM ratio of dose and distributopn volume

sumk = k12+k21+kel;
D = sqrt(sumk^2 - 4*k21*kel);



a = 0.5*(sumk + D);
b = 0.5*(sumk - D);



C= d_v2*k12/(a-b)*( exp(-b*(t-t0) ) - exp(-a*(t-t0)) );

tcoc = (t >= t0);

c2 = C.*tcoc;
